### PR TITLE
NMS-13684: Document Jetty SSL Setup

### PR DESCRIPTION
**Pull request by mmahacek**
### All Contributors

* [X] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [X] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-13684

### CODE REVIEW

**Review by indigo423 — CHANGES_REQUESTED** [review]
To be copy & paste friendly, we remove the leading `$` from the console inputs.

**Comment by mmahacek** [comment]
> To be copy & paste friendly, we remove the leading `$` from the console inputs.

Okay.   This is still an early draft that I'm working on cleaning up.

File: docs/modules/reference/pages/configuration/ssl.adoc [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+Alias:importkey  Password:importkey
+root@opennms:/home/monitor/ssl#
+----
+
+Now we have a proper JKS containing our private key and certificate in a file called keystore.ImportKey, using 'importkey' as alias and also as password
+
+Copy /root/keystore.ImportKey to /ect/opennms/opennms.keystore
+
+Change password importkey to changeit (this password is also in opennms.properties file, section https)
+`
```
Comment (indigo423) [inline docs/modules/reference/pages/configuration/ssl.adoc:329]: It looks like there are some markdown leftovers for the console in- and outputs.

**Review by indigo423 — CHANGES_REQUESTED** [review]

**Comment by indigo423** [comment]
> > To be copy & paste friendly, we remove the leading `$` from the console inputs.
> 
> Okay. This is still an early draft that I'm working on cleaning up.

ok got it, probably was too early :)

File: docs/modules/operation/pages/admin/https/https-server.adoc [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+This article explains how to configure OpenNMS' built-in Jetty web server to support HTTPS.
+
+TIP: It is typically easier to leave the Jetty config as default and use nginx or Apache as a reverse proxy.
+We have a https://opennms.discourse.group/t/how-to-use-nginx-as-ssl-proxy-with-opennms-horizon/208[Discourse forum post] with an example of using a reverse proxy.
+
+== Customize jetty.xml
+
+By default, no Jetty config file is included in {page-component-title}.
+There is a sample `jetty.xml` config in `$\{OPENNMS_HOME}/etc/examples` that can be used as a template.
+Copy this to `$\{OPENNMS_HOME}/etc` and remove comment tags for the section "Add HTTPS Support"
```
Comment (Bonrob2) [inline docs/modules/operation/pages/admin/https/https-server.adoc:14]:
```suggestion
Copy this to `$\{OPENNMS_HOME}/etc` and remove comment tags for the section "Add HTTPS Support".
```

File: docs/modules/operation/pages/admin/https/https-server.adoc [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+== Create a new Java keystore
+
+Using the keytool utility that ships as `$\{JAVA_HOME}/bin/keytool` with most Java distributions, create a new keystore and populate it with a new key.
+For the first question ("What is your first and last name"), enter the fully-qualified domain name by which people will be accessing your OpenNMS server's web UI.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/admin/https/https-server.adoc:19]:
```suggestion
For the first question ("What is your first and last name"), enter the fully-qualified domain name by which people will access your OpenNMS server's web UI.
```

File: docs/modules/operation/pages/admin/https/https-server.adoc [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+Choose this name correctly, as you will have to start over if you ever need to change it.
+Answer the remaining questions according to the specifics of your organization and locality.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/admin/https/https-server.adoc:21]:
```suggestion
Answer the remaining questions according to the specifics of your organization and location.
```

File: docs/modules/operation/pages/admin/https/https-server.adoc [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+Be sure to specify an appropriate number of days for the validity parameter.
+After this number of days elapses, the key you are generating will expire and you may no longer be able to use it to create new certificates.
+The example below specifies 731 days, which will make the key valid for two years (accounting for a possible leap year).
```
Comment (Bonrob2) [inline docs/modules/operation/pages/admin/https/https-server.adoc:25]:
```suggestion
The example below specifies 731 days, which makes the key valid for two years (accounting for a possible leap year).
```

File: docs/modules/operation/pages/admin/https/https-server.adoc [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  [Unknown]:  California
+What is the two-letter country code for this unit?
+  [Unknown]:  US
+Is CN=opennms.example.org, OU=Network Management Division, O=The Example Organization, L=Marina del Rey, ST=California, C=US correct?
+  [no]:  yes
+----
+
+== Restrict access to the plain-HTTP listener
+
+The steps to configure OpenNMS to start a Jetty HTTPS listener on port 8443 do not disable the plain HTTP listener that is present by default on port 8980.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/admin/https/https-server.adoc:57]:
```suggestion
The steps to configure OpenNMS to start a Jetty HTTPS listener on port 8443 do not disable the plain-HTTP listener that is present by default on port 8980.
```

**Comment by mmahacek** [comment]
@indigo423 Can you clear your requested changes on this PR to get it into the next release?

**Pull request merged**